### PR TITLE
vcflib: fixed build on arm64

### LINKED ISSUE #1463: Error installing vcflib
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

Output of `brew install brewsci/bio/vcflib` is the following.

```
==> Cloning https://github.com/ekg/vcflib.git
Updating /Users/alex/Library/Caches/Homebrew/vcflib--git
==> Checking out tag v1.0.1
HEAD is now at d150a89 Add C++0x badge
HEAD is now at d150a89 Add C++0x badge
Entering 'fastahack'
Entering 'filevercmp'
Entering 'fsom'
Entering 'googletest'
Entering 'intervaltree'
Entering 'libVCFH'
Entering 'multichoose'
Entering 'smithwaterman'
Entering 'tabixpp'
Synchronizing submodule url for 'htslib'
Entering 'tabixpp/htslib'
/Users/alex/Library/Caches/Homebrew/vcflib--git/fastahack
/Users/alex/Library/Caches/Homebrew/vcflib--git/filevercmp
/Users/alex/Library/Caches/Homebrew/vcflib--git/fsom
/Users/alex/Library/Caches/Homebrew/vcflib--git/googletest
/Users/alex/Library/Caches/Homebrew/vcflib--git/intervaltree
/Users/alex/Library/Caches/Homebrew/vcflib--git/libVCFH
/Users/alex/Library/Caches/Homebrew/vcflib--git/multichoose
/Users/alex/Library/Caches/Homebrew/vcflib--git/smithwaterman
/Users/alex/Library/Caches/Homebrew/vcflib--git/tabixpp
/Users/alex/Library/Caches/Homebrew/vcflib--git/tabixpp/htslib
==> Installing vcflib from brewsci/bio
==> make
Last 15 lines from /Users/alex/Library/Logs/Homebrew/vcflib/01.make:
gcc-11 -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c
gcc-11 -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c
gcc-11 -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c
gcc-11 -g -Wall -O2 -I.  -c -o kstring.o kstring.c
gcc-11 -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c
gcc-11 -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c
gcc-11 -g -Wall -O2 -I.  -c -o errmod.o errmod.c
gcc-11 -g -Wall -O2 -I.  -c -o faidx.o faidx.c
gcc-11 -g -Wall -O2 -I.  -c -o hfile.o hfile.c
gcc-11 -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c
gcc-11 -g -Wall -O2 -I.  -c -o hts.o hts.c
ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o
ranlib libhts.a
g++-11 -g -Wall -O2 -fPIC  -o tabix++ main.cpp tabix.o -I/private/tmp/vcflib-20211226-94802-xp20zo/tabixpp/htslib -L. -L/private/tmp/vcflib-20211226-94802-xp20zo/tabixpp/htslib -lhts -lbz2 -lm -lz -llzma -pthread \
		-lhts -lpthread -lm -lbz2 -llzma -lz

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/brewsci/homebrew-bio/issues
```

From the build output file, it looks like this is the problem.

```
fatal error: emmintrin.h: No such file or directory
```

[**Full Logs**](https://gist.github.com/9961e53891539e4c67f660bb4491b1ac)

### PR DESCRIPTION
**Pull request by alexreg**
Fixes #1463.

- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by alexreg** [comment]
Hmm, not sure why the Linux build is failing...

File: Formula/vcflib.rb [hunk 1]
```diff
@@ -2,32 +2,55 @@ class Vcflib < Formula
...
+  patch :DATA
 
   def install
-    system "make"
-    pkgshare.install Dir["bin/*.R"]
-    pkgshare.install Dir["bin/*.r"]
-    rm Dir["bin/*.R"]
-    rm Dir["bin/*.r"]
-    bin.install Dir["bin/*"]
+    cd "tabixpp" do
```
Comment (sjackman) [inline Formula/vcflib.rb:26]:
```
Errno::ENOENT: No such file or directory @ dir_chdir - tabixpp
```
https://github.com/brewsci/homebrew-bio/runs/4698088786?check_suite_focus=true#step:5:287

The directory `tabixpp` does not exist.
See https://github.com/vcflib/vcflib

Comment (alexreg) [reply Formula/vcflib.rb:26]: @sjackman Odd, as it seems to be building fine on macOS (both locally and in the CI). Just not Linux. Anyway, I see [this commit](https://github.com/ekg/tabixpp) which is relevant. How can I get the formula to check out tabixpp (say, the `v1.1.0` tag) in the root of the vcflib repo? Or should we create a separate formula for tabixpp?

Comment (sjackman) [reply Formula/vcflib.rb:26]: https://github.com/brewsci/homebrew-bio/runs/4698088830?check_suite_focus=true#step:4:285
```
  Error: An exception occurred within a child process:
285
    Errno::ENOENT: No such file or directory @ dir_chdir - tabixpp
286
  
287
==> brew audit brewsci/bio/vcflib --online --git --skip-style
288
Warning: install failed
289
==> SKIPPED brewsci/bio/vcflib
290
```
It seems to be failing on macOS as well in CI, though CI is reporting the run as successful for some strange unknown reason, possibly related to this `SKIPPED brewsci/bio/vcflib` message.

When testing locally, ensure that you've set `HOMEREW_NO_AUTO_UPDATE=1` to ensure that you're testing your changes. The auto update can `git stash` your changes when you run `brew install vcflib`.

Comment (alexreg) [reply Formula/vcflib.rb:26]: @sjackman Okay, thanks. Anyway, do you think we should create a separate formula for tabixpp? I can submit a PR for that too, if so.

Comment (sjackman) [reply Formula/vcflib.rb:26]: Yes, I think so. Is `tabixpp` a dependency of `vcflib`?

**Comment by sjackman** [comment]
If the primary purpose of this PR is to add support to `vcflib` for ARM64, it'd be better to submit your PR instead upstream to https://github.com/vcflib/vcflib

File: Formula/vcflib.rb (hunk 1 above)
Comment (alexreg) [reply Formula/vcflib.rb:26]: It is.

**Comment by alexreg** [comment]
@sjackman I'll have a look.

**Comment by jianshu93** [comment]
How to update the brew/sci formula to have this PR work for vcflib?

Thanks,

Jianshu

**Comment by alexreg** [comment]
@jianshu93 I'll be updating this PR next week when the author of vcflib merges a couple of PRs upstream. Then it should be ready to use.

If you need to install vcflib before that point, maybe use this formula locally, or build it manually by doing roughly what the formula does.

**Comment by jianshu93** [comment]
many thanks！

Jianshu

**Comment by jianshu93** [comment]
Thanks！I will try to install and report back.

Jianshu

**Pull request merged**